wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -638,9 +666,58 @@ class PlatformRepo(val walletApplication: WalletApplication) {
                     dashPayProfileDaoAsync.insert(profile!!)
                 }
             }
+            queueContactsUpdatedListeners()
             log.info("updating contacts and profiles took $watch")
         } catch (e: Exception) {
-            log.error("error updating contacts ${e.message}")
+            log.error(formatExceptionMessage("error updating contacts", e))
+        } finally {
+            updatingContacts.set(false)
+        }
+    }
+
+    // Create the Handler object (on the main thread by default)
+    val timerHandler = Handler()
+    var timerStarted = false
+
+    // Define the code block to be executed
+    private val executeUpdateContacts = object : Runnable {
+        override fun run() {
+            // Do something here on the main thread
```

Comment:
It looks like this and the comment below can be removed.

Reply:
https://github.com/dashevo/dash-wallet/pull/460/commits/929801ebf4d37deb29e65188d61588f5ddc7c7f1